inc/admin/menus/class-sidebar.php

```diff
@@ -728,10 +728,13 @@ private function getContextSlug( string $page, bool $is_main_site_page ): string
 	}
 
 	public static function handleH5pMenu(): void {
-		$user = get_user_by( 'id', get_current_user_id() );
+		$user_id = get_current_user_id();
+		if ($user_id === 0) {
+			return;
+		}
+		$user = get_user_by( 'id', $user_id );
 		if (
-			is_plugin_active( 'h5p/h5p.php' ) &&
-			in_array( 'subscriber', $user->roles, true )
+			$user && is_plugin_active( 'h5p/h5p.php' ) && !is_super_admin($user->ID) && $user->roles === ['subscriber']
```

Comment:
Clarified checks. user must exists, H5P must be active, ensure the user is not a super admin/network manager and that the ONLY role they have in the book is subscriber. We have weird situations where some users are both administrators and subscribers in the same book -- this will ensure that the menu is not removed for them.

Reply:
that makes sense. The only thing I'd change is that we don't need to check for the $user here, it is safe to assume the user exists at this point since it was checked here: https://github.com/pressbooks/pressbooks/pull/3937#discussion_r1978361722.
I'll push a friendly commit

Reply:
another think I'll propose is to check the roles with `in_array`, since `WP_User` object returns the role as an array: https://github.com/WordPress/wordpress-develop/blob/8a52d746e9bb85604f6a309a87d22296ce1c4280/src/wp-includes/class-wp-user.php#L80

Reply:
Yes, it can be an array, but we only care if the array has a single value and it is subscriber. That's why I checked for `['subscriber']` -- the brackets indicate an array, right?

Reply:
This way we don't have to worry if there are multiple array values (in the case of a user who accidentally has multiple roles). We're only proceeding if it's a single item array with the value of 'subscriber'

Reply:
I suppose we could also do something like $user->roles[0] === 'subscriber' but I thought this might be simpler

Reply:
yes, I've decided to leave as it is, since it makes sense.